src/transformers/modeling_flax_pytorch_utils.py

```diff
@@ -148,21 +152,24 @@ def is_key_or_prefix_key_in_dict(key: Tuple[str]) -> bool:
 def convert_pytorch_state_dict_to_flax(pt_state_dict, flax_model):
     # convert pytorch tensor to numpy
     # numpy currently does not support bfloat16, need to go over float32 in this case to not lose precision
-    try:
-        import torch  # noqa: F401
-    except (ImportError, ModuleNotFoundError):
-        logger.error(
-            "Loading a PyTorch model in Flax, requires both PyTorch and Flax to be installed. Please see"
-            " https://pytorch.org/ and https://flax.readthedocs.io/en/latest/installation.html for installation"
-            " instructions."
-        )
-        raise
 
-    weight_dtypes = {k: v.dtype for k, v in pt_state_dict.items()}
-    pt_state_dict = {
-        k: v.numpy() if not v.dtype == torch.bfloat16 else v.float().numpy() for k, v in pt_state_dict.items()
-    }
+    def numpify_state_dict(state_dict):
+        from_bin = is_torch_available() and isinstance(next(iter(state_dict.values())), torch.Tensor)
+        if from_bin:
+            bfloat16_dtype = torch.bfloat16
+        else:
+            bfloat16_dtype = "bfloat16"
+
+        state_dict = {}
+        for k, v in pt_state_dict.items():
+            if v.dtype == bfloat16_dtype:
+                state_dict[k] = v.float().numpy() if from_bin else v
+            else:
+                state_dict[k] = v.numpy() if from_bin else v
+
+        return state_dict
```

Comment:
@Narsil, curious if you have something a bit cleaner than that in mind to do dtype analysis across safetensors + pt

Reply:
Not really torch.bfloat16 is really singular compared to numpy dtypes (which is a custom dtype for flax/tf)